src/views/search/components/AdvancedSearchModal/AdvancedSearchModal.tsx

```diff
@@ -0,0 +1,239 @@
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import {
+  Button,
+  DatePicker,
+  DatePickerProps,
+  Form,
+  FormGroup,
+  InputGroup,
+  InputGroupItem,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  Split,
+  SplitItem,
+  TextInput,
+} from '@patternfly/react-core';
+
+import { AdvancedSearchInputs } from '../../utils/types';
+
+import LabelsMultiSelect from './components/LabelsMultiSelect';
+import MemoryUnitSelect from './components/MemoryUnitSelect';
+import NumberInput from './components/NumberInput';
+import NumberOperatorSelect, { NumberOperator } from './components/NumberOperatorSelect';
+import ProjectMultiSelect from './components/ProjectMultiSelect';
+
+const initialVCPU = {
+  operator: NumberOperator.GreaterThan,
+  value: NaN,
+};
+
+const initialMemory = {
+  operator: NumberOperator.GreaterThan,
+  unit: CAPACITY_UNITS.GiB,
+  value: NaN,
+};
+
+type AdvancedSearchModalProps = Pick<ModalComponentProps, 'isOpen' | 'onClose'> & {
+  onSubmit: (searchInputs: Required<AdvancedSearchInputs>) => void;
+  prefillInputs?: AdvancedSearchInputs;
+};
+
+const AdvancedSearchModal: FC<AdvancedSearchModalProps> = ({
+  isOpen,
+  onClose,
+  onSubmit,
+  prefillInputs = {},
+}) => {
+  const { t } = useKubevirtTranslation();
+
+  const [name, setName] = useState(prefillInputs.name ?? '');
+  const [projects, setProjects] = useState(prefillInputs.projects ?? []);
+  const [description, setDescription] = useState(prefillInputs.description ?? '');
+  const [labels, setLabels] = useState(prefillInputs.labels ?? []);
+  const [ip, setIP] = useState(prefillInputs.ip ?? '');
+  const [dateFrom, setDateFrom] = useState<Date>();
+  const [dateFromString, setDateFromString] = useState(prefillInputs.dateCreatedFrom ?? '');
+  const [dateToString, setDateToString] = useState(prefillInputs.dateCreatedTo ?? '');
+  const [vCPU, setVCPU] = useState(prefillInputs.vCPU ?? initialVCPU);
+  const [memory, setMemory] = useState(prefillInputs.memory ?? initialMemory);
+
+  const [isValidDate, setIsValidDate] = useState(true);
+
+  const isEmptyForm = useMemo<boolean>(
+    () =>
+      name === '' &&
+      projects.length === 0 &&
+      description === '' &&
+      labels.length === 0 &&
+      ip === '' &&
+      dateFromString === '' &&
+      dateToString === '' &&
+      isNaN(vCPU.value) &&
+      isNaN(memory.value),
+    [
+      name,
+      projects,
+      description,
+      labels,
+      ip,
+      dateFromString,
+      dateToString,
+      vCPU.value,
+      memory.value,
+    ],
+  );
+
+  const onDateChange: (
+    setDateString: (dateString: string) => void,
+    setDate?: (date: Date) => void,
+  ) => DatePickerProps['onChange'] = (setDateString, setDate) => (_, dateString, date) => {
+    if (date) {
+      setDateString(dateString);
+      setDate?.(date);
+    } else {
+      setDateString('');
+      setDate?.(undefined);
+    }
+  };
+
+  const resetForm = () => {
```

Comment:
useCallbkack is needed here?

Reply:
`resetForm` function doesn't need React `useCallback` because its only outer context dependencies are `setName` etc. React `useState` setters, which are guaranteed to be referentially stable across all re-renders.

https://react.dev/reference/react/useState

> The `set` function has a stable identity, so you will often see it omitted from Effect dependencies, but including it will not cause the Effect to fire. If the linter lets you omit a dependency without errors, it is safe to do. [Learn more about removing Effect dependencies.](https://react.dev/learn/removing-effect-dependencies#move-dynamic-objects-and-functions-inside-your-effect)